app/src/main/java/com/example/indoor_volleyball/Fragments/GymFinderFragment.java

```diff
@@ -58,20 +49,30 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         return view;
     }
 
+    //Sub-classes have get data method that's it.
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        //setContentView(R.layout.fragment_gym_finder);
-        rvGyms = binding.rvGyms;
-        allGymsByDistance = new ArrayList<>();
-        gymsFollowed = new ArrayList<>();
-        adapterAllGyms = new GymAdapter(getContext(), allGymsByDistance);
-        adapterUserGyms = new UserGymAdapter(getContext(), gymsFollowed);
-        rvGyms.setAdapter(adapterAllGyms);
-        rvGyms.setLayoutManager(new LinearLayoutManager(getContext()));
-        allGymsByDistance(ParseUser.getCurrentUser().getParseGeoPoint("longLat"));
+
+        PagerAdapter pagerAdapter = new PagerAdapter(this);
+        binding.vpGymList.setAdapter(pagerAdapter);
+        TabLayout tlGymsList = binding.tlGymsList;
+        new TabLayoutMediator(tlGymsList, binding.vpGymList,
+                (tab, position) -> tab.setText(pagerAdapter.getTitle(position))).attach();
+
+
+//        rvGyms = binding.rvGyms;
+//        allGymsByDistance = new ArrayList<>();
+//        gymsFollowed = new ArrayList<>();
+//        adapterAllGyms = new GymAdapter(getContext(), allGymsByDistance);
+//        adapterUserGyms = new UserGymAdapter(getContext(), gymsFollowed);
+//        rvGyms.setAdapter(adapterAllGyms);
+//        rvGyms.setLayoutManager(new LinearLayoutManager(getContext()));
+//        allGymsByDistance(ParseUser.getCurrentUser().getParseGeoPoint("longLat"));
+
 
         //TODO refresh listener.
+        // FIXME ggfdghjhf
 //        //swipeContainer = (SwipeRefreshLayout) view.findViewById(R.id.swipeContainer);
```

Comment:
Remove all this.  Swipe refresh layout would go in the GymListFragment we created.